utils/config.go

```diff
@@ -688,11 +688,6 @@ func GenerateClientConfig(c *model.Config, diagnosticId string, license *model.L
 			props["DataRetentionEnableFileDeletion"] = strconv.FormatBool(*c.DataRetentionSettings.EnableFileDeletion)
 			props["DataRetentionFileRetentionDays"] = strconv.FormatInt(int64(*c.DataRetentionSettings.FileRetentionDays), 10)
 		}
-
```

Comment:
I think we can have it in both client config generators. For example, if we enable the Terms of Service while a user is logged in, it will need that information in the configuration (I think).

Reply:
`GenerateClientConfig` calls `GenerateLimitedClientConfig` and then we add in the missing fields so there is no duplication.

https://github.com/mattermost/mattermost-server/pull/9989/files#diff-5308e96aebc56970c6e69b0ce769be11R515